### PR TITLE
[Bulky] Amend existing booking.

### PR DESCRIPTION
**Pull request by dracos**
Fixes https://github.com/mysociety/societyworks/issues/3742 [skip changelog]
This lets staff amend a bulky waste booking. The amendment 'moderates' the report and resends it, plus adds a cancellation report for the original Bartec ID.

### CODE REVIEW

File: perllib/FixMyStreet/App/Form/Waste/Bulky/Amend.pm [hunk 1]
```diff
@@ -0,0 +1,295 @@
+package FixMyStreet::App::Form::Waste::Bulky::Amend;
+
+use utf8;
+use DateTime::Format::Strptime;
+use HTML::FormHandler::Moose;
+use JSON::MaybeXS;
+extends 'FixMyStreet::App::Form::Waste';
+
+has_page intro => (
+    title => 'Book bulky goods collection',
```
Comment (davea) [inline perllib/FixMyStreet/App/Form/Waste/Bulky/Amend.pm:10]: Should this be `Amend`?

File: perllib/FixMyStreet/App/Form/Waste/Bulky/Amend.pm [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+    type => 'Submit',
+    value => 'Amend booking',
+    element_attr => { class => 'govuk-button' },
+    order => 999,
+);
+
+has_field chosen_date => (
+    type                 => 'Select',
+    widget               => 'RadioGroup',
+    label                => 'Available dates',
```
Comment (davea) [inline perllib/FixMyStreet/App/Form/Waste/Bulky/Amend.pm:119]: This shows the booked date twice:

image

**Comment by dracos** [comment]
"Summary page says you can't amend booking, but that might not be true any more:" - I guess as this is staff only, might want to keep this message.
"The dropdowns on the items page flash open with a single item and disappear the first time they're clicked (subsequent clicks are OK):" - that happens now with normal booking if you go back from the summary page, wonder what's happening there.
"It's possible to "amend" the booking by clicking through the form and making no changes, which resends it and raises a cancellation. Should we ensure at least one change is made?" - yeah, makes sense.
"If I sign in as the member of public the booking was made for, I can see the amend buttons on the bin days and summary pages but they just take me to the bin days page:" - whoops.

Will look at photos, thought I'd fixed that.

**Review by davea — APPROVED** [review]
Woo, works perfectly!

**Comment by dracos** [comment]
The dropdown issue (that existed before this PR) - think I've worked out how to fix that too now.

**Pull request merged**